common/common.go

```diff
@@ -15,28 +14,34 @@ import (
 	"github.com/zcash-hackworks/lightwalletd/walletrpc"
 )
 
-func GetSaplingInfo(rpcClient *rpcclient.Client) (int, int, string, string, error) {
-	result, rpcErr := rpcClient.RawRequest("getblockchaininfo", make([]json.RawMessage, 0))
-
-	var err error
-	var errCode int64
-
-	// For some reason, the error responses are not JSON
-	if rpcErr != nil {
-		errParts := strings.SplitN(rpcErr.Error(), ":", 2)
-		errCode, err = strconv.ParseInt(errParts[0], 10, 32)
-		if err == nil && errCode == -8 {
-			return -1, -1, "", "", nil
-		}
-		return -1, -1, "", "", errors.Wrap(rpcErr, "error requesting block")
-	}
-
+func GetSaplingInfo(rpcClient *rpcclient.Client, log *logrus.Entry) (int, int, string, string) {
+	// This request must succeed or we can't go on; give zcashd time to start up
 	var f interface{}
-	err = json.Unmarshal(result, &f)
-	if err != nil {
-		return -1, -1, "", "", errors.Wrap(err, "error reading JSON response")
+	retryCount := 0
+	for {
+		result, rpcErr := rpcClient.RawRequest("getblockchaininfo", make([]json.RawMessage, 0))
+		if rpcErr == nil {
+			if retryCount > 0 {
+				log.Warn("getblockchaininfo RPC successful")
```

Comment:
Note that if we weren't successful on our first attempt (`retryCount > 0`), that means we logged some errors... so now that we're finally successful, let's indicate that in the log file (otherwise it's unclear that we're okay now). We could log this every time (not just if `retryCount > 0`), but it's really only confusing not to if we've had earlier errors.